packages/devtools/web/index.html

```diff
@@ -36,13 +36,51 @@
                 );
             }
         })();
+
+        // Plotly wrappers.
+        function hookupPlotlyClick(domName, jsFunction) {
+            var graph = document.getElementById(domName);
+            graph.on('plotly_click', jsFunction);
+        }
+
+        function _copyTrace(x, y, orgTrace, xData, yData, traces) {
+          if (x.length > 0) {
+            var newX = [];
+            var newY = [];
+            for (var idx = 0; idx < x.length; idx++) {
+              newX.push(x[idx]);
+              newY.push(y[idx]);
+            }
+            xData.push(newX);
+            yData.push(newY);
+            traces.push(orgTrace);
+          }
+        }
+
+        function myExtendTraces(domName, x0, x1, x2, x3, y0, y1, y2, y3, orgTraces) {
```

Comment:
it would be nice to just call
Plotly.extendTraces 
directly from dart code.
What is blocking that?

Reply:
I think if reified types but not positive.  plotly is checkin that all axes be the same base and they're not.  Not sure if I need an abstract class for this in the interop then maybe it would work.  I add a comment I'd like to eliminate (the commented code you see in frames_bar_plotly is the original code.

Reply:
fine to leave as is with a note that this code is written in JS to work around an edge case. 
In general, I would also suggest passing in x and y as arrays with 4 elements rather than as 
x0, x1, x2, x3, y0,y1,y2,y3